public/app/features/alerting/unified/Silences.test.tsx

```diff
@@ -285,48 +252,20 @@ describe('Silence edit', () => {
       await waitFor(() => expect(ui.editor.timeRange.get()).toHaveTextContent(startDateString));
       await waitFor(() => expect(ui.editor.timeRange.get()).toHaveTextContent(endDateString));
 
-      await userEvent.type(ui.editor.matcherName.get(), 'foo');
-      await userEvent.type(ui.editor.matcherOperatorSelect.get(), '=');
-      await userEvent.tab();
-      await userEvent.type(ui.editor.matcherValue.get(), 'bar');
-
-      // TODO remove skipPointerEventsCheck once https://github.com/jsdom/jsdom/issues/3232 is fixed
-      await userEvent.click(ui.editor.addMatcherButton.get(), { pointerEventsCheck: PointerEventsCheckLevel.Never });
-      await userEvent.type(ui.editor.matcherName.getAll()[1], 'bar');
-      await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[1], '!=');
-      await userEvent.tab();
-      await userEvent.type(ui.editor.matcherValue.getAll()[1], 'buzz');
-
-      // TODO remove skipPointerEventsCheck once https://github.com/jsdom/jsdom/issues/3232 is fixed
-      await userEvent.click(ui.editor.addMatcherButton.get(), { pointerEventsCheck: PointerEventsCheckLevel.Never });
-      await userEvent.type(ui.editor.matcherName.getAll()[2], 'region');
-      await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[2], '=~');
-      await userEvent.tab();
-      await userEvent.type(ui.editor.matcherValue.getAll()[2], 'us-west-.*');
-
-      // TODO remove skipPointerEventsCheck once https://github.com/jsdom/jsdom/issues/3232 is fixed
-      await userEvent.click(ui.editor.addMatcherButton.get(), { pointerEventsCheck: PointerEventsCheckLevel.Never });
-      await userEvent.type(ui.editor.matcherName.getAll()[3], 'env');
-      await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[3], '!~');
-      await userEvent.tab();
-      await userEvent.type(ui.editor.matcherValue.getAll()[3], 'dev|staging');
+      await enterSilenceLabel(0, 'foo', MatcherOperator.equal, 'bar');
+
+      await addAdditionalMatcher();
+      await enterSilenceLabel(1, 'bar', MatcherOperator.notEqual, 'buzz');
+
+      await addAdditionalMatcher();
+      await enterSilenceLabel(2, 'region', MatcherOperator.regex, 'us-west-.*');
+
+      await addAdditionalMatcher();
+      await enterSilenceLabel(3, 'env', MatcherOperator.notRegex, 'dev|staging');
 
       await userEvent.click(ui.editor.submit.get());
 
-      await waitFor(() =>
```

Comment:
I've removed some of the specific assertions here as we were checking that we called the API in a certain way. Instead I'm checking that the table is then rendered again afterwards - anyone have some other ideas of something to assert against here? I was trying to avoid continuing a pattern where we checked what MSW received 🤷

Reply:
We need to migrate these `api` calls assertions to check UI changes after API actions

Reply:
I added it back in by checking the server events but I'm not keen on using that approach much, so I marked the new helper method as deprecated immediately to make it clear in usages 😁 

Ideally we would have a stateful server for the tests (using MSW Data or something), and then we can assert its on the page correctly after without caring about the API's received body, or we could use cache updates in RTKQ and assert on the page state after that